test/e2e_node/topology_manager_test.go

```diff
@@ -122,9 +201,79 @@ func configureTopologyManagerInKubelet(f *framework.Framework, policy string) {
 		framework.ExpectNoError(err)
 		return nodes == 1
 	}, time.Minute, time.Second).Should(gomega.BeTrue())
+
+	return newCfg.ReservedSystemCPUs
 }
 
-func runTopologyManagerSuiteTests(f *framework.Framework) {
+// getSRIOVDevicePluginPod returns the Device Plugin pod for sriov resources in e2e tests.
+func getSRIOVDevicePluginPod() *v1.Pod {
+	ds := readDaemonSetV1OrDie(testfiles.ReadOrDie(SRIOVDevicePluginDSYAML))
+	p := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      SRIOVDevicePluginName,
+			Namespace: metav1.NamespaceSystem,
+		},
+
+		Spec: ds.Spec.Template.Spec,
+	}
+
+	return p
+}
+
+func readConfigMapV1OrDie(objBytes []byte) *v1.ConfigMap {
+	v1.AddToScheme(appsScheme)
+	requiredObj, err := runtime.Decode(appsCodecs.UniversalDecoder(v1.SchemeGroupVersion), objBytes)
+	if err != nil {
+		panic(err)
+	}
+	return requiredObj.(*v1.ConfigMap)
+}
+
+func readServiceAccountV1OrDie(objBytes []byte) *v1.ServiceAccount {
+	v1.AddToScheme(appsScheme)
+	requiredObj, err := runtime.Decode(appsCodecs.UniversalDecoder(v1.SchemeGroupVersion), objBytes)
+	if err != nil {
+		panic(err)
+	}
+	return requiredObj.(*v1.ServiceAccount)
+}
+
+func findSRIOVResource(node *v1.Node) (string, int64) {
+	re := regexp.MustCompile(`^intel.com/.*sriov.*`)
```

Comment:
SR-IOV device plugin also supports Mellanox devices. I think it would be a good idea if this could be expanded to work with either Intel or Mellanox.

Reply:
Yup, already improved here: https://github.com/kubernetes/kubernetes/pull/87645/commits/0fad14ffe1c3daadf2a6d47c3ed68e8fd8e80bfb